### ISSUE TITLE
Capturing additional AFDS claims beyond the standard Django auth user model

### ISSUE THREAD

**Issue by jdmills-edu** [jobec/django-auth-adfs#24]
If I wanted to send additional claims (e.g. employee number, department, title, device name, that sort of thing) and affiliate them with the Django auth user that your package creates and authenticates, how would you recommend that I go about doing that? I've seen several guides for extending the vanilla Django auth user model, but what would be the best way to streamline the process with django_auth_adfs?

**Comment by jobec** [comment]
Mapping claims to a user’s properties is possible by configuring this setting: http://django-auth-adfs.readthedocs.io/en/latest/configuration.html#claim-mapping

If you need a boolean field to be set, use this setting: http://django-auth-adfs.readthedocs.io/en/latest/configuration.html#boolean-claim-mapping

**Comment by jdmills-edu** [comment]
Understood. But I'm talking about fields that don't exist in the Django auth user model. For example, if I just add a claim, let's call it "myClaim", and map it in the settings, I'll get this message.

> User model has no field named 'myClaim'. Check ADFS claims mapping.

What would be the best way to extend the default user model in a way that captures these claims as the come into django_auth_adfs?

**Comment by jobec** [comment]
These are 2 separate things.

First you need to figure out how to extend the default user model with extra fields. This article seems to explain is quite well: https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html#abstractuser

Next, when you create a model with extra fields, you'll need to map whatever claims you configured on the ADFS server to the fields in your user model. As explained in the `CLAIM_MAPPING` setting's documentation

**Comment by jdmills-edu** [comment]
I had already seen the doc you linked and was started down a much more complicated path. But I clicked it anyway and it took me to the section "Extending User Model Using a Custom Model Extending AbstractUser".

> Custom User Model from AbstractUser: The way Django handles auth is a perfect fit for your project but still you want to add extra attributes without having to create a separate Model.

That sounded like what I was trying to do, so I gave it a shot. And then I mapped the additional claims in the CLAIM_MAPPING section, like you said. And of course, it worked!

THANK YOU!
